pkg/apis/certmanager/v1/types_certificate.go

```diff
@@ -138,10 +138,9 @@ type CertificateSpec struct {
 	// denoted issuer.
 	SecretName string `json:"secretName"`
 
-	// SecretTemplate defines annotations and labels to be propagated
-	// to the Kubernetes Secret when it is created or updated. Once created,
-	// labels and annotations are not yet removed from the Secret when they are
-	// removed from the template. See https://github.com/jetstack/cert-manager/issues/4292
+	// SecretTemplate defines annotations and labels to be copied to the
+	// Certificate's Secret. Labels and annotations on the Secret will be changed
+	// as they appear on the SecretTemplate.
```

Comment:
Currently, the issuing controller creates a Secret with many annotations:

```yaml
apiVersion: v1
kind: Secret
metadata:
  name: example
  annotations:
    cert-manager.io/alt-names: example.now.mael-valais-gcp.jetstacker.net
    cert-manager.io/certificate-name: example
    cert-manager.io/common-name: example.now.mael-valais-gcp.jetstacker.net
    cert-manager.io/ip-sans: ""
    cert-manager.io/issuer-group: cert-manager.io
    cert-manager.io/issuer-kind: Issuer
    cert-manager.io/issuer-name: letsencrypt
    cert-manager.io/uri-sans: ""
type: kubernetes.io/tls
data:
  tls.crt: ...
  tls.key: ...
```

Does the sentence:

> Labels and annotations on the Secret will be changed as they appear on the SecretTemplate

mean that the above annotations will be overridden by the `secretTemplate` annotations? Or are they appended to the existing annotations?

Reply:
It doesn't mean they will be overridden, they will be appended.

I'll update the comment on the API type.